In [1]:
import pandas as pd 
import matplotlib.pyplot as plt

In [2]:
file_name= "3M-5-[2-5000]-[DEPARTEMENT]-lcm.out"
plot_name = file_name.replace('out','html')

# Sankey : 
file:///home/abdelouahab/PFE/lcm-tests/plots/3M-5-[2-5000]-[DEPARTEMENT]-lcm.html

In [75]:
groups = pd.read_csv(f"plots/groups/{file_name}",index_col=0)
links = pd.read_csv(f"plots/links/{file_name}",index_col=0)

In [76]:
groups.head()

user_ids  support           itemsets  \
0  [187833 322520 427250 518210 576233 603204]        6   4190175 19419902   
1         [206078 263990 365536 622027 752568]        5    5461017 7445935   
2  [428202 602649 622027 752568 754685 908778]        6          6027 6026   
3         [204029 286729 288846 525249 610249]        5   3461024 18064205   
4         [584447 602649 622027 752568 754685]        5      18064205 6026   

       period  property_values  depth  size  DEPARTEMENT  
0  2017-02-01               59    0.0     6           59  
1  2017-02-01               78    0.0     5           78  
2  2017-02-01               78    0.0     6           78  
3  2017-02-01               78    0.0     5           78  
4  2017-02-01               78    0.0     5           78

In [77]:
links.head()

target user_id
source                
0           48  187833
0          174  427250
1           26  622027
1           30  622027
1           31  752568

### Merge query

In [84]:
target_users= set('107030 108938 110899 114666 118478'.split())
target_users

{'107030', '108938', '110899', '114666', '118478'}

In [85]:
groups[groups.user_ids.apply(lambda x : set(target_users).issubset(x[1:-1].split()) )]

user_ids  support  \
259  [ 107030  108938  110899  114666  118478  1277...       60   
280  [ 107030  108938  110899  114666  118478  1277...       60   
293  [ 107030  108938  110899  114666  118478  1277...       54   
306  [ 107030  108938  110899  114666  118478  1277...       48   

          itemsets      period  property_values     depth  size  DEPARTEMENT  
259   12520268 451  2018-11-01               59  0.636364    60           59  
280   451 12520268  2019-02-01               59  0.727273    60           59  
293   12520268 451  2019-05-01               59  0.818182    54           59  
306   12520268 451  2019-08-01               59  0.909091    48           59

### Split query

In [1]:
# TODO 

# Promotion queries 

### Finding promotion products


In [3]:
products_description = pd.read_csv("datasets/Total/items.csv",sep=";",encoding="latin-1")
products_description.head()

ARTICLE_ID             DESCRIPTION
0  3425909001007  RECHARGE LAVAGE TW - A
1  3425909001083          REC LAVAGE PJE
2  3124480186188   SCHWEPPES MOJITO 50CL
3  3559750000893         NEOCLEAN JANTES
4  5410041005707     LU PIMS ORANGE 150G

In [7]:
def find_all(x,keywords):
    for i in keywords:
        if i.lower() not in x.lower():
            return False
    return True
products_description[products_description["DESCRIPTION"].apply(lambda x : find_all(x,["anti",""]))]

ARTICLE_ID                    DESCRIPTION
132    3217770310513                TOTAL ANTI CREV
196    3011449248578  TW.DEMOUST. ANTI FIENTE 400ML
272    3597610235511                GEL ANTISEPTIQU
342    5026349023963         Rainix anti-buée 200ml
629    5410076309757                AMBI PUR ANTI T
...              ...                            ...
18946  3513765119988       TAPIS ANTI DERAP. POUR T
19040  5908241631173                AROMA ANTI TOBA
19204  7612720841721       ARBRE MAGIQUE  CLIP ANTI
19252  3011449301402          TW DEMOUST ANTI 500ML
19635  3222471609316    CHIANTI DOCG CL.SOMMEL.75CL

[114 rows x 2 columns]

3158849232001	BIDON AD BLUE 10L
3158849282006	ADBLUE BIDON 5L
3158849282006	ADBLUE 5L BIDON
7070091000701	TOTAL ADBLUE 10L
3158849232001	BIDON AD BLUE 10L
3217770310902	TOTAL ADBLUE 10L
3011441109006	TOTAL ADBLUE BIDON 10L
3425901018140	TOTAL ADBLUE BIDON 10L
3425901018140	TOTAL ADBLUE 10L
7070091000701	TOTAL ADBLUE BIDON 10L
3245901018140	TOTAL ADBLUE 10L
3011441109006	TOTAL ADBLUE 10L
3507020013302	TOTAL ADBLUE 10L
3660963045141	TOTAL ADBLUE BIDON 10L
1051790999999	ADBLUE 10L
3460537800028	TOTAL ADBLUE 10L
3245901018140	TOTAL ADBLUE BIDON 10L
3158849232001	Bidon Ad Blue 10l Synchro
3460531000004	TOTAL ADBLUE BIDON 10L
4042865227723	ADBLUE 10L
4042865227723	ADBLUE 10L - BASH
723003       	TOTAL ADBLUE 10L
8717953106902	ADBLUE 10L
3323727100219	ADBLUE 10L AD
3701239300011	ADBLUE 10L
8717306130615	TOTAL ADBLUE 10L
3158849232001	TOTAL ADBLUE 10L
8717306130615	TOTAL ADBLUE BIDON 10L
3158849232001	TOTAL ADBLUE BIDON 10L
723003      	TOTAL ADBLUE BIDON 10L
3540709003157	ADBLUE 10L
3660963045141	TOTAL ADBLUE 10L
8717953106902	AD Blue 10L
3507020013302	TOTAL ADBLUE BIDON 10L
3425901018409	TOTAL ADBLUE 10L
3158849282006	ADBLUE BIDON 5L
3158849282006	ADBLUE 5L BIDON

3267025010606	ELF EVOLUTION 5W30 2 L
3267025000836	ELF EVOLUTION SXR 5W40 1L
3267025001406	ELF EVOLUTION C
3267021013236	ELF EVOLUTION SXR 5W30 1
3267021013229	ELF EVOLUTION SXR 5W30 2L
3267025001406	ELF EVOLUTION CRV 0W30 1L
3267021013236	ELF EVOLUTION SXR 5W30 1L
3267025017650	ELF EVOLUTION 2L 5W30
3267025011375	ELF EVOLUTION 500 TS

3267025007262	ELF MOTO 2 OFF ROAD 1L 10W40 SAE 30
3267021020531	ELF MOTO 4 HP ECO 10W40 1L
3267021020432	ELF MOTO 4 XT TECH 10W50 1L
3011449257075	ELF POLISH MOTO 400ML
3267021020432	ELF MOTO 4 XT TECH 10W50
3267025007460	ELF MOTO 4 TECH
3011449257037	ELF MOTO LUSTRE
3267021030837	ELF MOTO 2 SELF MIX 1L
3267025000072	ELF MOTO GEAR OIL 10W40 1L
3267025007262	ELF MOTO 2 OFF ROAD 1L
3267025007231	ELF MOTO 2 SELF MIX 1L
3011449257082	ELF SHAMPOING MOTO 400ML
3267021030738	ELF MOTO 2 DX RATIO 1 L
3011449257051	ELF NET.JANTES MOTO 400ML
3267025007262	ELF MOTO 2 OFF
3267025013072	ELF MOTO BRAKE FLUID DOT 5-1
3267025007361	ELF MOTO 4 ROAD 1L 10W40
3267025007361	ELF MOTO 4 ROAD
3267025007361	ELF MOTO 4 ROAD 10W40 1L
3267025013072	ELF MOTO BRAKE
3267025007293	ELF MOTO 2 TECH 1L SAE 30
3011449257044	ELF DEGRAISS. MOTO 400ML
3267021030530	ELF MOTO 2 XT TECH 10W40 1L
3267025007460	ELF MOTO 4 TECH 10W50 1L
3267025007293	ELF MOTO 2 TECH
3267025013065	ELF MOTO BRAKE FLUID DOT4
3011449262581	ELF NET.JANTES MOTO 50ML
3267025000027	ELF MOTO CHAINLUB 400ML
3267025007293	ELF MOTO 2 TECH 1L

3425901018195	TOTAL ACTIVA INEO ECS 5W30 5L
3425901018720	TOTAL ACTIVA 9000 5W40 2L
3425900000436	TOTAL ACTIVA 90
3425901018195	TOTAL ACTIVA IN
3425900000436	TOTAL ACTIVA 9000 5W40 2L
3425901014951	TOTAL ACTIVA 9000 FUTUR NFC E 5W30 5L
3425901015729	TOTAL ACTIVA 9000 5W40 2L
3425901015729	TOTAL ACTIVA 90
3425901018720	TOTAL ACTIVA 90
3425901027494	TOTAL ACTIVA 90
3425901018201	TOTAL ACTIVA IN


3425901029320	TOTAL QUARTZ 0W30 5L
3425901025872	TOTAL QUARTZ 90
3425901025872	TOTAL QUARTZ 9000 FUTUR 5W30 1L NFC
3425901029290	TOTAL QUARTZ IN
3425901019505	TOTAL QUARTZ INEO MC3 5W30
3425901019505	TOTAL QUARTZ IN
3425901017976	TOTAL QUARTZ E. 9000 0W30 5L
3425901027869	TOTAL QUARTZ 9000 FUEL E
3425901028170	TOTAL QUARTZ INEO LONG LIFE 5W30 1L
3425901028170	TOTAL QUARTZ IN
3425901029290	TOTAL QUARTZ INEO FIRST PSA 0W30
3425901019734	TOTAL QUARTZ 10W-40 1L
3425901029320	TOTAL QUARTZ 0W
3425901018232	TOTAL QUARTZ IN


3068320055022	EVIAN 6X50CL
3461025	        CR EVIAN PET 50CL
99999112	    CE EVIAN BTTLE PVC 50CL
3068320110783	EVIAN 50CL PET
3068320115214	EVIAN 50CL PET
8461014         Re Evian Pet 50Cl
3068320055008	EVIAN 50CL PET
3068320099651	EVIAN 50CL
3068320112312	EVIAN 50CL
3068320115214	Evian 50cl
7461024	        Lun Evian Pet 50Cl
3068320110783	EVIAN 50CL
3068320099651	Evian 50cl
3068320112312	Evian 50cl
3068320099651	EVIAN 50CL PET
3068320055008	EVIAN 50CL
3068320115214	EVIAN 50CL
3068320112312	EVIAN 50CL PET
3068320055008	Evian 50cl
3068320110783	Evian 50cl


3217770310513	TOTAL ANTI CREV

In [12]:
pd.DataFrame(promo_products).


0
0    3158849232001
1    3158849282006
2    3158849282006
3    7070091000701
4    3158849232001
..             ...
116  3068320115214
117  3068320112312
118  3068320055008
119  3068320110783
120  3217770310513

[121 rows x 1 columns]

In [20]:
df = pd.DataFrame(promo_products,columns=["id","name"])
df.to_csv('datasets/Total/promo_products.csv',index=None)

In [21]:
!cat datasets/Total/promo_products.csv

id,name
3158849232001,BIDON AD BLUE 10L
3158849282006,ADBLUE BIDON 5L
3158849282006,ADBLUE 5L BIDON
7070091000701,TOTAL ADBLUE 10L
3158849232001,BIDON AD BLUE 10L
3217770310902,TOTAL ADBLUE 10L
3011441109006,TOTAL ADBLUE BIDON 10L
3425901018140,TOTAL ADBLUE BIDON 10L
3425901018140,TOTAL ADBLUE 10L
7070091000701,TOTAL ADBLUE BIDON 10L
3245901018140,TOTAL ADBLUE 10L
3011441109006,TOTAL ADBLUE 10L
3507020013302,TOTAL ADBLUE 10L
3660963045141,TOTAL ADBLUE BIDON 10L
1051790999999,ADBLUE 10L
3460537800028,TOTAL ADBLUE 10L
3245901018140,TOTAL ADBLUE BIDON 10L
3158849232001,Bidon Ad Blue 10l Synchro
3460531000004,TOTAL ADBLUE BIDON 10L
4042865227723,ADBLUE 10L
4042865227723,ADBLUE 10L - BASH
723003,TOTAL ADBLUE 10L
8717953106902,ADBLUE 10L
3323727100219,ADBLUE 10L AD
3701239300011,ADBLUE 10L
8717306130615,TOTAL ADBLUE 10L
3158849232001,TOTAL ADBLUE 10L
8717306130615,TOTAL ADBLUE BIDON 10L
3158849232001,TOTAL ADBLUE BIDON 10L
723003,TOTAL ADBLUE BIDON 10L
3540709003157,ADBLUE 10L
3660963045141,